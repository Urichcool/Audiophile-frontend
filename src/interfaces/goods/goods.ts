export interface IGoods {
  _id: string;
  slug: string;
  name: string;
  image: { mobile: string; tablet: string; desktop: string };
  category: string;
  categoryImage: { mobile: string; tablet: string; desktop: string };
  new: boolean;
  price: number;
  description: string;
  features: string;
  includes: { quantity: number; item: string; _id:string }[];
  gallery: {
    first: { mobile: string; tablet: string; desktop: string };
    second: { mobile: string; tablet: string; desktop: string };
    third: { mobile: string; tablet: string; desktop: string };
  };
  previewImage: {
    mobile: string;
    tablet: string;
    desktop: string;
  };
}
