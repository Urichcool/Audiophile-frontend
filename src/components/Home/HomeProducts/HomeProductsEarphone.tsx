import { FC } from "react";
import { IGoods } from "../../../interfaces/goods/goods";
import Button2 from "../../Reusable-Components/Buttons/Button2";
import Loader from "../../Reusable-Components/Loader";
import { AnimationOnScroll } from "react-animation-on-scroll";
import Picture from "../../Reusable-Components/Picture";

interface IHomeProductsEarphoneProps {
  data: IGoods | undefined;
  isFetching: boolean;
}

const HomeProductsEarphone: FC<IHomeProductsEarphoneProps> = ({
  data,
  isFetching,
}) => {
  return (
    <div className="home-products-earphone-container">
      <div className="home-products-earphone-picture-container">
        {isFetching ? (
          <Loader />
        ) : (
          <AnimationOnScroll
            animateIn="animate__fadeIn"
            delay={1000}
            animateOnce
            offset={0}
          >
            <Picture
              imageSrc={{
                desktop: data?.previewImage.desktop,
                tablet: data?.previewImage.tablet,
                mobile: data?.previewImage.mobile,
              }}
              sizes={{
                desktopWidth: "270",
                desktopHeight: "300",
                tabletWidth: "220",
                tabletHeight: "250",
                mobileWidth: "160",
                mobileHeight: "180",
                }}
                name={data?.name}
                className=""
            />
          </AnimationOnScroll>
        )}
      </div>
      <div className="home-products-earphone-content-container">
        {isFetching ? (
          <Loader />
        ) : (
          <AnimationOnScroll
            animateIn="animate__fadeInDown"
            delay={1500}
            animateOnce
            offset={0}
          >
            <div className="home-products-earphone-content">
              <h3 className="H4-manrope-bold  home-products-earphone-title">
                {data?.name}
              </h3>
              <Button2 id={data?._id} />
            </div>
          </AnimationOnScroll>
        )}
      </div>
    </div>
  );
};

export default HomeProductsEarphone;
