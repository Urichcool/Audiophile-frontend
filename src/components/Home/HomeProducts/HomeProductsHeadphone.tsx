import { FC } from "react";
import { IGoods } from "../../../interfaces/goods/goods";
import Button2 from "../../Reusable-Components/Buttons/Button2";
import Loader from "../../Reusable-Components/Loader";
import { AnimationOnScroll } from "react-animation-on-scroll";

interface IHomeProductsHeadphoneProps {
  data: IGoods | undefined;
  isFetching: boolean;
}

const HomeProductsHeadphone: FC<IHomeProductsHeadphoneProps> = ({
  data,
  isFetching,
}) => {
  return (
    <div className="home-products-headprone-container">
      {isFetching ? (
        <Loader />
      ) : (
        <>
          <AnimationOnScroll
            animateIn="animate__fadeIn"
            delay={1000}
            animateOnce
            offset={0}
          >
            <picture className="home-products-headphone-picture">
              <source
                srcSet={data?.previewImage.desktop}
                media="(min-width: 1440px)"
                width={"270"}
                height={"300"}
              />
              <source
                srcSet={data?.previewImage.desktop}
                media="(min-width: 768px)"
                width={"220"}
                height={"250"}
              />
              <img
                width={"200"}
                height={"250"}
                src={data?.previewImage.desktop}
                alt={data?.name}
              />
            </picture>
          </AnimationOnScroll>
          <AnimationOnScroll
            animateIn="animate__fadeInDown"
            delay={1500}
            animateOnce
            offset={0}
          >
            <div className="home-products-headprone-content">
              <h3 className="H4-manrope-bold home-products-headprone-title">
                {data?.name}
              </h3>
              <Button2 />
            </div>
          </AnimationOnScroll>
        </>
      )}
    </div>
  );
};

export default HomeProductsHeadphone;
