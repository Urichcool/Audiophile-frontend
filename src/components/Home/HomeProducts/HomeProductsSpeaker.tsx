import { FC } from "react";
import { IGoods } from "../../../interfaces/goods/goods";
import Loader from "../../Reusable-Components/Loader";
import { shortDescrFunc } from "../../../utils/shortDescrFunc";
import Button4 from "../../Reusable-Components/Buttons/Button4";
import { AnimationOnScroll } from "react-animation-on-scroll";
import Picture from "../../Reusable-Components/Picture";

interface IHomeProductsSpeakerProps {
  data: IGoods | undefined;
  isFetching: boolean;
}

const HomeProductsSpeaker: FC<IHomeProductsSpeakerProps> = ({
  data,
  isFetching,
}) => {
  return (
    <div
      className="home-products-speakers-container"
      data-testid={"home-products-speaker-section"}
    >
      {isFetching ? (
        <Loader />
      ) : (
        <>
          <AnimationOnScroll
            animateIn="animate__fadeIn"
            delay={2000}
            animateOnce
            style={{ zIndex: "2", position: "relative" }}
            offset={0}
          >
            <Picture
              imageSrc={{
                desktop: data?.previewImage.desktop,
                tablet: data?.previewImage.tablet,
                mobile: data?.previewImage.mobile,
              }}
              sizes={{
                desktopWidth: "410",
                desktopHeight: "493",
                tabletWidth: "197",
                tabletHeight: "237",
                mobileWidth: "172",
                mobileHeight: "207",
              }}
              name={data?.name}
              className={"home-products-speakers-picture"}
            />
          </AnimationOnScroll>
          <AnimationOnScroll
            animateIn="animate__fadeIn"
            delay={500}
            animateOnce
            offset={0}
          >
            <div className="home-products-speakers-circle1" />
          </AnimationOnScroll>
          <AnimationOnScroll
            animateIn="animate__fadeIn"
            delay={1000}
            animateOnce
            offset={0}
          >
            <div className="home-products-speakers-circle2"></div>
          </AnimationOnScroll>
          <AnimationOnScroll
            animateIn="animate__fadeIn"
            delay={1500}
            animateOnce
            offset={0}
          >
            <div className="home-products-speakers-circle3"></div>
          </AnimationOnScroll>
          <AnimationOnScroll
            animateIn="animate__fadeInDown"
            delay={2500}
            animateOnce
            offset={150}
          >
            <div className="home-products-speakers-content">
              <h2 className="H1-manrope-bold home-products-speakers-title">
                {data?.name}
              </h2>
              <p className="Body-manrope-medium home-products-speakers-description">
                {shortDescrFunc(data?.description)}
              </p>
              <Button4 id={data?._id} />
            </div>
          </AnimationOnScroll>
        </>
      )}
    </div>
  );
};

export default HomeProductsSpeaker;
