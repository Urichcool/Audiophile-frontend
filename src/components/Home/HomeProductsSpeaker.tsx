import { FC } from "react";
import { IGoods } from "../../interfaces/goods/goods";
import Loader from "../Reusable-Components/Loader";
import { shortDescrFunc } from "../../utils/shortDescrFunc";
import Button4 from "../Reusable-Components/Buttons/Button4";

interface IHomeProductsSpeakerProps {
  data: IGoods | undefined;
  isFetching: boolean;
}

const HomeProductsSpeaker: FC<IHomeProductsSpeakerProps> = ({
  data,
  isFetching,
}) => {
  return (
    <div className="home-products-speakers-container">
      {isFetching ? (
        <Loader />
      ) : (
        <>
          <picture className="home-products-speakers-picture">
            <source
              srcSet={data && data.previewImage.desktop}
              media="(min-width: 1440px)"
              width={"410"}
              height={"493"}
            />
            <source
              srcSet={data && data.previewImage.desktop}
              media="(min-width: 768px)"
              width={"197"}
              height={"237"}
            />
            <img
              src={data && data.previewImage.desktop}
              alt={data && data.name}
              width={"172"}
              height={"207"}
            />
          </picture>
          <div className="home-products-speakers-circle1"></div>
          <div className="home-products-speakers-circle2"></div>
          <div className="home-products-speakers-circle3"></div>
          <div className="home-products-speakers-content">
            <h2 className="H1-manrope-bold home-products-speakers-title">
              {data?.name}
            </h2>
            <p className="Body-manrope-medium home-products-speakers-description">
              {shortDescrFunc(data?.description)}
            </p>
            <Button4 />
          </div>
        </>
      )}
    </div>
  );
};

export default HomeProductsSpeaker;
