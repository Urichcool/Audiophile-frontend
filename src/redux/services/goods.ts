import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import { IGoods } from "../../interfaces/goods/goods";
import { RootState } from "../store";

export const goodsApi = createApi({
  reducerPath: "goodsApi",
  baseQuery: fetchBaseQuery({
    baseUrl: "https://audiophile-backend.onrender.com/",
  }),
  endpoints: (builder) => ({
    getNewGoods: builder.query<IGoods, void>({
      query: () => "goods/new",
    }),
    getOtherGoods: builder.query<IGoods[], void>({
      query: () => "goods/other",
    }),
    getGoodsById: builder.query<IGoods, string>({
      query: (id) => `goods/${id}`,
    }),
    getAllHeadphones: builder.query<IGoods[], void>({
      query: () => "headphones",
    }),
    getAllEarphones: builder.query<IGoods[], void>({
      query: () => "earphones",
    }),
    getAllSpeakers: builder.query<IGoods[], void>({
      query: () => "speakers",
    }),
    getGoodsStock: builder.query<{ stock: number }, string>({
      query: (id) => `stock/check/${id}`,
    }),
    checkGoodsCartStock: builder.mutation<
      { isEnoughCartStock: boolean },
      { id: string; quantity: number }[]
    >({
      query: (cart) => ({
        url: `stock/check/cart`,
        body: cart,
        method: "POST",
      }),
    }),
  }),
});

export const {
  useGetNewGoodsQuery,
  useGetOtherGoodsQuery,
  useGetGoodsByIdQuery,
  useGetAllHeadphonesQuery,
  useGetAllEarphonesQuery,
  useGetAllSpeakersQuery,
  useGetGoodsStockQuery,
  useCheckGoodsCartStockMutation
} = goodsApi;
