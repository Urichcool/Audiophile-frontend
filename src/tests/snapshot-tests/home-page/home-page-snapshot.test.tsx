import HomeProductsEarphone from "../../../components/Home/HomeProducts/HomeProductsEarphone";
import { IGoods } from "../../../interfaces/goods/goods";
import { renderWithReduxAndRouter } from "../../helpers/RenderWithReduxAndRouter";
import { screen } from "@testing-library/react";
import { homeProductSectionTestProps } from "./component-props";
import HomeProductsHeadphone from "../../../components/Home/HomeProducts/HomeProductsHeadphone";
import HomeProductsSpeaker from "../../../components/Home/HomeProducts/HomeProductsSpeaker";

describe("home-page-snapshot-tests", () => {
  test("home-products-earphone-section-should-match-snapshot", () => {
    const { data, isFetching }: { data: IGoods; isFetching: boolean } =
      homeProductSectionTestProps;
    renderWithReduxAndRouter(
      <HomeProductsEarphone data={data} isFetching={isFetching} />
    );
    const homeProductsEarphoneSection: HTMLDivElement = screen.getByTestId(
      "home-products-earphone-section"
    );
    expect(homeProductsEarphoneSection).toMatchSnapshot();
  });
  test("home-products-headphone-section-should-match-snapshot", () => {
    const { data, isFetching }: { data: IGoods; isFetching: boolean } =
      homeProductSectionTestProps;
    renderWithReduxAndRouter(
      <HomeProductsHeadphone data={data} isFetching={isFetching} />
    );
    const homeProductsHeadphoneSection: HTMLDivElement = screen.getByTestId(
      "home-products-headphone-section"
    );
    expect(homeProductsHeadphoneSection).toMatchSnapshot();
  });
  test("home-products-speaker-section-should-match-snapshot", () => {
    const { data, isFetching }: { data: IGoods; isFetching: boolean } =
      homeProductSectionTestProps;
    renderWithReduxAndRouter(
      <HomeProductsSpeaker data={data} isFetching={isFetching} />
    );
    const homeProductsSpeakerSection: HTMLDivElement = screen.getByTestId(
      "home-products-speaker-section"
    );
    expect(homeProductsSpeakerSection).toMatchSnapshot();
  });
});
