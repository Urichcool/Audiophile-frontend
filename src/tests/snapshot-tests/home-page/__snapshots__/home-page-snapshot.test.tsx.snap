// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home-page-snapshot-tests home-products-earphone-section-should-match-snapshot 1`] = `
<div
  class="home-products-earphone-container"
  data-testid="home-products-earphone-section"
>
  <div
    class="home-products-earphone-picture-container"
  >
    <div
      class="animate__animated"
      style="animation-duration: 1s; opacity: 0;"
    >
      <picture
        class=""
        data-testid="picture-component"
      >
        <source
          height="300"
          media="(min-width: 1440px)"
          srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443229/audiophile/goods/product-yx1-earphones/desktop/image-product-removebg-desktop-removebg-preview_kvis0b.png"
          width="270"
        />
        <source
          height="250"
          media="(min-width: 768px)"
          srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443285/audiophile/goods/product-yx1-earphones/tablet/image-product-removebg-tablet-removebg-preview_baskgx.png"
          width="220"
        />
        <img
          alt="YX1 Wireless Earphones"
          height="180"
          src="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443263/audiophile/goods/product-yx1-earphones/mobile/image-product-removebg-mobile-removebg-preview_rrnkle.png"
          width="160"
        />
      </picture>
    </div>
  </div>
  <div
    class="home-products-earphone-content-container"
  >
    <div
      class="animate__animated"
      style="animation-duration: 1s; opacity: 0;"
    >
      <div
        class="home-products-earphone-content"
      >
        <h3
          class="H4-manrope-bold  home-products-earphone-title"
        >
          YX1 Wireless Earphones
        </h3>
        <button
          class="button2"
        >
          see product
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`home-page-snapshot-tests home-products-headphone-section-should-match-snapshot 1`] = `
<div
  class="home-products-headprone-container"
  data-testid="home-products-headphone-section"
>
  <div
    class="animate__animated"
    style="animation-duration: 1s; opacity: 0;"
  >
    <picture
      class="home-products-headphone-picture"
      data-testid="picture-component"
    >
      <source
        height="300"
        media="(min-width: 1440px)"
        srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443229/audiophile/goods/product-yx1-earphones/desktop/image-product-removebg-desktop-removebg-preview_kvis0b.png"
        width="270"
      />
      <source
        height="250"
        media="(min-width: 768px)"
        srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443285/audiophile/goods/product-yx1-earphones/tablet/image-product-removebg-tablet-removebg-preview_baskgx.png"
        width="220"
      />
      <img
        alt="YX1 Wireless Earphones"
        height="250"
        src="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443263/audiophile/goods/product-yx1-earphones/mobile/image-product-removebg-mobile-removebg-preview_rrnkle.png"
        width="200"
      />
    </picture>
  </div>
  <div
    class="animate__animated"
    style="animation-duration: 1s; opacity: 0;"
  >
    <div
      class="home-products-headprone-content"
    >
      <h3
        class="H4-manrope-bold home-products-headprone-title"
      >
        YX1 Wireless Earphones
      </h3>
      <button
        class="button2"
      >
        see product
      </button>
    </div>
  </div>
</div>
`;

exports[`home-page-snapshot-tests home-products-speaker-section-should-match-snapshot 1`] = `
<div
  class="home-products-speakers-container"
  data-testid="home-products-speaker-section"
>
  <div
    class="animate__animated"
    style="animation-duration: 1s; opacity: 0; z-index: 2; position: relative;"
  >
    <picture
      class="home-products-speakers-picture"
      data-testid="picture-component"
    >
      <source
        height="493"
        media="(min-width: 1440px)"
        srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443229/audiophile/goods/product-yx1-earphones/desktop/image-product-removebg-desktop-removebg-preview_kvis0b.png"
        width="410"
      />
      <source
        height="237"
        media="(min-width: 768px)"
        srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443285/audiophile/goods/product-yx1-earphones/tablet/image-product-removebg-tablet-removebg-preview_baskgx.png"
        width="197"
      />
      <img
        alt="YX1 Wireless Earphones"
        height="207"
        src="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443263/audiophile/goods/product-yx1-earphones/mobile/image-product-removebg-mobile-removebg-preview_rrnkle.png"
        width="172"
      />
    </picture>
  </div>
  <div
    class="animate__animated"
    style="animation-duration: 1s; opacity: 0;"
  >
    <div
      class="home-products-speakers-circle1"
    />
  </div>
  <div
    class="animate__animated"
    style="animation-duration: 1s; opacity: 0;"
  >
    <div
      class="home-products-speakers-circle2"
    />
  </div>
  <div
    class="animate__animated"
    style="animation-duration: 1s; opacity: 0;"
  >
    <div
      class="home-products-speakers-circle3"
    />
  </div>
  <div
    class="animate__animated"
    style="animation-duration: 1s; opacity: 0;"
  >
    <div
      class="home-products-speakers-content"
    >
      <h2
        class="H1-manrope-bold home-products-speakers-title"
      >
        YX1 Wireless Earphones
      </h2>
      <p
        class="Body-manrope-medium home-products-speakers-description"
      >
        Tailor your listening experience with bespoke dynamic drivers from the new YX1 Wireless Earphones. Enj...
      </p>
      <button
        class="button4"
      >
        See Product
      </button>
    </div>
  </div>
</div>
`;
